src/main/kotlin/org/axonframework/intellij/ide/plugin/usage/AggregateAnnotator.kt

```diff
@@ -0,0 +1,44 @@
+package org.axonframework.intellij.ide.plugin.usage
+
+import com.intellij.lang.annotation.AnnotationHolder
+import com.intellij.lang.annotation.Annotator
+import com.intellij.lang.annotation.HighlightSeverity
+import com.intellij.psi.PsiElement
+import org.axonframework.intellij.ide.plugin.api.AxonAnnotation
+import org.axonframework.intellij.ide.plugin.util.isAggregate
+import org.jetbrains.uast.UClass
+import org.jetbrains.uast.UIdentifier
+import org.jetbrains.uast.toUElement
+
+/**
+ * Shows a warning on classes annotated with `@Aggregate` when:
+ * - They have no zero-argument constructor. This constructor is required by Axon Framework to construct the aggregate.
+ * - They have no AggregateIdentifier field. This is required by Axon Framework.
+ *
+ */
+class AggregateAnnotator : Annotator {
```

Comment:
Are there any other useful annotations? Requirements of Axon that can be easily missed for new people?

Ideas:
- Assigning the aggregateIdentifier in EventSourcingHandler of event created by constructor
- Entities with an `@EntityId` field need an `AggregateMember` annotations
... others? 

Any case, let's make issues for them after merging this PR.

Reply:
I like both ideas as nice to haves actually. 
Next to that, I could think off warning for usage of the `@EventHandler` annotation instead of the `@EventSourcingHandler` annotation.

Note that the `@EntityId` annotation is only required whenever the aggregate member annotation is used for a group of entities. If there's a singular entity, Axon will automatically route commands to it without the use of an `@EntityId` annotated field.

Reply:
Added, any Event Sourcing specific stuff should only happen whenever the Aggregate is actually being Event Sourced. As soon as it's State-Stored, those things will not apply.